### ISSUE TITLE
Found a way to at least partially add search mode

### ISSUE THREAD

**Issue by ghost** [goosepirate/lox365#17]
Hello,
The reason I want to use XLOOKUP is to perform a reverse search starting at the last item, so I modified your code to add that part. I've only added it in the vertical direction but hopefully this will help you make a version that works for vertical too. I also didn't add the binary search modes. Here is the modified code for the lox365.py file:
```
def XLOOKUP(lookupValue, lookupArray, returnArray, ifNotFound=ERR_NA, searchMode=1):
    if ifNotFound is None: ifNotFound = ERR_NA
    if searchMode is None: searchMode = 1
    lookup_direction = 0 # 0 is vertical; 1 is horizontal
    if len(lookupArray) == 1 and len(lookupArray[0]) > 1: lookup_direction = 1
    try:
        if searchMode == -1:
            if lookup_direction == 0:
                return (returnArray[len(returnArray) - lookupArray[::-1].index((lookupValue,)) - 1],)
            if lookup_direction == 1:
                return tuple((row[lookupArray[0].index(lookupValue)],) for row in returnArray)
        else:
            if lookup_direction == 0:
                return (returnArray[lookupArray.index((lookupValue,))],)
            if lookup_direction == 1:
                return tuple((row[lookupArray[0].index(lookupValue)],) for row in returnArray)
    except ValueError: return ((ifNotFound,),)
```

Here is the added node to the addin.xcu file:
```
            <node oor:name="searchMode" oor:op="replace">
              <prop oor:name="DisplayName"><value xml:lang="en">[Search mode]</value></prop>
              <prop oor:name="Description"><value xml:lang="en">The order of the search. 1=Start from first item (default) -1=Reverse.</value></prop>
            </node>
```

**Comment by goosepirate** [comment]
Hi, nice work! Would you be willing to add the reverse search in the horizontal direction? If so I will add it to the code.

**Comment by ovari** [comment]
XLOOKUP has been added to LibreOffice 24.8 as shown at https://wiki.documentfoundation.org/ReleaseNotes/24.8#Calc

Can you please test that the reverse search works for you in both the vertical and horizontal directions using LibreOffice 24.8?

Thank you

**Comment by Nekkowe** [comment]
> as shown at https://wiki.documentfoundation.org/ReleaseNotes/24.8#Calc
![image](https://github.com/goosepirate/lox365/assets/28980049/7048dd4e-8223-467d-aa25-3e24693e8029)

It doesn't say that there, though?

**Comment by ovari** [comment]
@Nekkowe it was shown in the Release notes; however, it seems that has been removed and will be re-added later.

The XLOOKUP function has been implemented as shown in https://bugs.documentfoundation.org/show_bug.cgi?id=127293#c39

You can test in the daily builds. Links to the daily builds are shown in the LibreOffice issue tracker.

For your information, [XMATCH](https://bugs.documentfoundation.org/show_bug.cgi?id=128425) and [FILTER, SORT, SORTBY, UNIQUE, SEQUENCE, RANDARRAY functions](https://bugs.documentfoundation.org/show_bug.cgi?id=159872) are being added too.

**Comment by Nekkowe** [comment]
Ooooh, nice!! I've really been looking forward to some of these..! Thank you for the information!

**Comment by ovari** [comment]
> Ooooh, nice!! I've really been looking forward to some of these..! Thank you for the information!

Can you please merge https://github.com/goosepirate/lox365/pull/20? Hopefully this can help discover, test and provide feedback to lox365 users?

Thank you

**Comment by Nekkowe** [comment]
Me? I'm afraid I'm not a maintainer, just a passerby, so I don't have write access to this repository... you'll want @goosepirate for that, I think.

**Comment by ovari** [comment]
@Nekkowe FYI below is the LibreOffice Help XLOOKUP Function link
https://help.libreoffice.org/master/en-US/text/scalc/01/func_xlookup.html

**Comment by ovari** [comment]
> Hi, nice work! Would you be willing to add the reverse search in the horizontal direction? If so I will add it to the code.

LibreOffice 24.8 implements the XLOOKUP function
https://bugs.documentfoundation.org/show_bug.cgi?id=127293

LibreOffice Help also includes the XLOOKUP function
https://help.libreoffice.org/master/en-US/text/scalc/01/func_xlookup.html

@goosepirate does LibreOffice 24.8 implementation of the XLOOKUP function add the reverse search in the vertical and horizontal directions?

Thank you

**Comment by ovari** [comment]
> > as shown at https://wiki.documentfoundation.org/ReleaseNotes/24.8#Calc
> > ![image](https://private-user-images.githubusercontent.com/28980049/307967313-7048dd4e-8223-467d-aa25-3e24693e8029.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk2NTY5MjUsIm5iZiI6MTcxOTY1NjYyNSwicGF0aCI6Ii8yODk4MDA0OS8zMDc5NjczMTMtNzA0OGRkNGUtODIyMy00NjdkLWFhMjUtM2UyNDY5M2U4MDI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI5VDEwMjM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYzAwNDczNjAzNGRhMjllODM4NDA0MmExZTM2ZmMxMDIxMDI1MGZhMDdmZDI5NjE2ODNkOTBmMzZlNzY4YjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v9WsEVgOGJG3WmLtm8_9Dzf61yuPlCHZ7wagPDQBry8)
> 
> It doesn't say that there, though?

It's there now, please see screenshot below.

![image](https://github.com/goosepirate/lox365/assets/17465872/0f9a6345-3771-4584-9c31-47094d9a2643)

**Comment by ovari** [comment]
OASIS [OFFICE-4154 add XLOOKUP function](https://issues.oasis-open.org/browse/OFFICE-4154) includes search mode which has been implemented in [Bug 127293 - Add XLOOKUP function in Calc](https://bugs.documentfoundation.org/show_bug.cgi?id=127293) and available in [LibreOffice Calc 24.8](https://wiki.documentfoundation.org/ReleaseNotes/24.8#Calc).

@goosepirate can this issue please be closed?

Thank you

**Comment by goosepirate** [comment]
Thanks. Closing.
